src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -271,11 +272,43 @@ public AzureAppConfigurationOptions Connect(Uri endpoint, TokenCredential creden
             }
 
             ConnectionString = null;
-            Endpoint = endpoint;
+            Endpoints = new List<Uri>() { endpoint };
             Credential = credential;
             return this;
         }
 
+        /// <summary>
+        /// Connect the provider to Azure App Configuration and their replicas using list of endpoints and token credential.
+        /// </summary>
+        /// <param name="endpoints">The list of endpoints of the Azure App Configuration and it's replicas to connect to.</param>
+        /// <param name="credential">Token credential to use to connect.</param>
+        public AzureAppConfigurationOptions Connect(IEnumerable<Uri> endpoints, TokenCredential credential)
+        {
+            if (endpoints == null || endpoints.Count() == 0)
+            {
+                throw new ArgumentNullException(nameof(endpoints));
+            }
+
+            IEnumerable<string> configStoreNames = endpoints.Select(endpoint => endpoint.Host.Split('.')[0]);
+            var primaryStores = configStoreNames.Where(storeName => !storeName.Contains(ConfigurationStoreConstants.ConfigStoreNameReplicaSeparator));
+
+            if (primaryStores.Count() > 1)
+            {
+                throw new ArgumentException($"{nameof(endpoints)} can have only one primary configuration store endpoint.");
+            }
+
+            if (!configStoreNames.All(storeName => storeName.Split(configStoreNameSplitSeparator, StringSplitOptions.None)[0].Equals(primaryStores.Single(), StringComparison.OrdinalIgnoreCase)))
```

Comment:
There is no guarantee the primary store is always in the endpoint list. The `primaryStores` could be empty.

Reply:
With the new implementation, this logic is no longer needed.